code/script.js

```diff
@@ -227,99 +258,166 @@ const setSecret = () => {
+        `Yes, the person has ${currentQuestion.value} hair! Keep all with hair ${currentQuestion.value} `
+      )
+    } else {
+      alert (
+        `No, the person doesn't have ${currentQuestion.value} hair! Remove all with hair ${currentQuestion.value} `
+      )
+      
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  if (keep) { //Ex: Does the person have glasses? If yes, they will be kept, if not, removed
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value) 
+    //Keeps everyone who has got the asked attribute
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value) 
+    //Removes everyone who doesn't have the asked attribute
+  }
+  
 
-  // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+// Once clicking guess, the player has to confirm the guess to avoid misstakes
+const guess = (suspect) => {
+  const guessConfirmation =  confirm(`Are you sure that you want to guess on ${suspect}?`)
+  if (guessConfirmation){
+    console.log(suspect)
+    checkMyGuess(suspect)
+  }
+  
+// If the player wants to guess, invoke the checkMyGuess function.
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+//this is what happens when you confirm your guess
+const checkMyGuess = (suspect) => {
+  if (suspect === secret.name){
+    //the H1 in the html
+    winOrLoseText.innerHTML = `
+    Congratulations! Your guess on ${suspect} is correct! Try playing again and see if you can win again!
+    `
+  } else {
+    winOrLoseText.innerHTML = `
+    Oh boy.. Your guess on ${suspect} is wrong.
+    Looks like you need more practice. Press "Play Again" and have another go!
+    `
+  }
+
+  winOrLose.style.display = 'flex' //shows if you win or lose
+  board.style.display = 'none' //this removes the board
+  
```

Comment:
This is also very nicely done. So smooth with just replacing the innerHTML with text depending on the outcome! And well done with show and hide the different sections.